### PR TITLE
Move the schema check logic to it’s own service and use it to control when the deployment.json is imported.

### PR DESCRIPTION
**Pull request by chirino**
Caching logic was not exactly right for when the cache did not hold the entire dataset (like when we restart and the dataset is in the DB but not in the cache).

### CODE REVIEW

File: dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java [hunk 1]
```diff
@@ -140,13 +142,14 @@ public String generatePK(Cache<String, WithId> entityMap) {
 
     @SuppressWarnings("unchecked")
     public <T extends WithId> ListResult<T> fetchAll(Class<T> model, Function<ListResult<T>, ListResult<T>>... operators) {
-        Kind kind = Kind.from(model);
```
Comment (jimmidyson) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: What are these changes about? Seem unrelated to the schema check/update service?

Comment (chirino) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: See commit message: 'Caching logic was not exactly right for when the cache did not hold the entire dataset (like when we restart and the dataset is in the DB but not in the cache).'

after restart fetchAll was returning back an empty list, since we were trying to only fetch from the cache.

Comment (jimmidyson) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: Having a PR that includes unrelated changes to the pr title, albeit with details in commit messages, doesn't ​make reviews easy... 

Once we get post demo we're going to work out better ways to do this.

**Comment by jimmidyson** [comment]
@iocanel Build isn't running on rh-idev so no status check for PRs... Any idea what's going on?

File: dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java [hunk 2]
```diff
@@ -140,13 +142,14 @@ public String generatePK(Cache<String, WithId> entityMap) {
...
-        Cache<String, WithId> cache = caches.getCache(kind.getModelName());
 
-        ListResult<T> result = ListResult.of((Collection<T>) cache.values());
-         
-        if( result == null ) {
+        ListResult<T> result;
+        if( getDataAccessObject(model)!=null ) {
```
Comment (jimmidyson) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: Isn't this going to ignore the cache if there is a dao for this kind? Not sure that's what we want to do.

Comment (chirino) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: Right.  This PR breaks the build without the cache changes.

Comment (chirino) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: Well it only ignores the caches for fetch all.  Single lookups should be cached.  At scale this is what you would want anyways right?

File: dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java (hunk 1 above)
Comment (chirino) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: Its not unrelated.  The caching changes have to go otherwise the tests would be breaking.

File: dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java (hunk 2 above)
Comment (jimmidyson) [inline dao/src/main/java/com/redhat/ipaas/dao/manager/DataManager.java]: Who knows what we want at scale? It all needs revisiting post-summit. I'd even consider a distributed cache that we use for persistence, potentially even obviating the need for PostgreSQL...

Regardless, right now we're not using the cache properly anyway so let's merge this and revisit it later.

**Comment by jimmidyson** [comment]
Manually merging...
